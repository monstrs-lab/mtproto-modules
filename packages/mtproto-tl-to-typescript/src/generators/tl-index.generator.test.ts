import { tmpdir }               from 'node:os'
import { join }                 from 'node:path'

import { describe }             from '@jest/globals'
import { expect }               from '@jest/globals'
import { it }                   from '@jest/globals'
import { Project }              from 'ts-morph'
import { ScriptTarget }         from 'ts-morph'
import { ModuleKind }           from 'ts-morph'
import { ModuleResolutionKind } from 'ts-morph'

import { TLIndexGenerator }     from './tl-index.generator.js'

describe('mtproto tl to typescript', () => {
  describe('tl index generator', () => {
    it('check generate index file', async () => {
      const project = new Project({
        compilerOptions: {
          outDir: tmpdir(),
          module: ModuleKind.ES2022,
          target: ScriptTarget.ES2022,
          moduleResolution: ModuleResolutionKind.Bundler,
        },
      })

      project.createSourceFile(join(tmpdir(), 'test.ts'))

      new TLIndexGenerator(project).generate()

      expect(project.getSourceFile(join(tmpdir(), 'index.ts'))?.getFullText()).toMatchSnapshot()
    })
  })
})
