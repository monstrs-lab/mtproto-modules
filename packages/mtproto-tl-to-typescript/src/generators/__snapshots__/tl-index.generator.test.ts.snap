// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mtproto tl to typescript tl index generator check generate index file 1`] = `
"export * from "./test.js";
"
`;
